artichoke-frontend depends on mruby backend internals

`airb` depends on:

- parser
- context

`string_scan_bench` depends on:

- global variable get

Add APIs to artichoke-core for these items.

Blocked on GH-116.

```
error[E0624]: method `inner` is private
  --> artichoke-frontend/src/bin/string_scan_bench.rs:45:83
   |
45 |     unsafe { sys::mrb_gv_set(mrb, interp.0.borrow_mut().sym_intern("$data"), data.inner()) }
   |
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -35,7 +35,7 @@ rustyline = { version = "9.1.2", optional = true, default-features = false }
 termcolor = { version = "1.1.0", optional = true }
 
 [dependencies.artichoke-backend]
-version = "0.8.0"
+version = "0.9.0"
 path = "artichoke-backend"
 default-features = false
 

--- a/artichoke-backend/Cargo.toml
+++ b/artichoke-backend/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "artichoke-backend"
-version = "0.8.0"
+version = "0.9.0"
 authors = ["Ryan Lopopolo <rjl@hyperbo.la>"]
 edition = "2021"
 rust-version = "1.58.1"

--- a/artichoke-backend/src/value.rs
+++ b/artichoke-backend/src/value.rs
@@ -80,11 +80,10 @@ impl Value {
         Self::default()
     }
 
-    /// The [`sys::mrb_value`] that this [`Value`] wraps.
-    // TODO(GH-251): make `Value::inner` pub(crate).
+    /// Retrieve the inner [`sys::mrb_value`] that this [`Value`] wraps.
     #[inline]
     #[must_use]
-    pub const fn inner(&self) -> sys::mrb_value {
+    pub(crate) const fn inner(&self) -> sys::mrb_value {
         self.0
     }
 

--- a/artichoke-backend/tests/integration/extension.rs
+++ b/artichoke-backend/tests/integration/extension.rs
@@ -15,7 +15,7 @@ unsafe extern "C" fn container_initialize(mrb: *mut sys::mrb_state, slf: sys::mr
     let inner = inner.try_convert_into::<i64>(&guard).unwrap_or_default();
     let container = Box::new(Container(inner));
     let result = Box::<Container>::box_into_value(container, slf, &mut guard).unwrap_or_default();
-    result.inner()
+    result.into()
 }
 
 unsafe extern "C" fn container_value(mrb: *mut sys::mrb_state, slf: sys::mrb_value) -> sys::mrb_value {
@@ -26,7 +26,7 @@ unsafe extern "C" fn container_value(mrb: *mut sys::mrb_state, slf: sys::mrb_val
     } else {
         Value::nil()
     };
-    result.inner()
+    result.into()
 }
 
 impl File for Container {

--- a/artichoke-backend/tests/integration/leak/boxing.rs
+++ b/artichoke-backend/tests/integration/leak/boxing.rs
@@ -30,7 +30,7 @@ unsafe extern "C" fn container_initialize(mrb: *mut sys::mrb_state, slf: sys::mr
     let container = Container(inner);
     let result = Container::box_into_value(container, slf, &mut guard);
     match result {
-        Ok(value) => value.inner(),
+        Ok(value) => value.into(),
         Err(exception) => error::raise(guard, exception),
     }
 }

